RDA support: Make MATE panel aware of being run inside a remote deskt…

sunweaver opened the pull request:
…op technology.

lukefromdc commented:
I do not have any remote desktop setups or experience for testing this. Someone who does needs to test this

vkareh commented:
I have x2go set up, I'll give this a try tomorrow at the office

raveit65 commented:
What is the benefit of this PR?
Steps to test?
What should happen?
What should not happen?
Better description please....
Sorry, my crystal ball is broken :)

sunweaver commented:
> What is the benefit of this PR?

This PR is the first PR of a series of PRs I plan to submit in the next days/weeks. For now, I have another PR planned against mate-screensaver and already actively work on it.

The PR series will make MATE more aware of its execution context. Meaning: MATE will become able to behave differently when run inside a remote desktop session compared to when run on local hardware. Different remote technology support different feature sets. These feature sets will also be queriable from within MATE components. All the remote desktop gnosis, I plan to pack into librda (Remote Desktop Awareness shared lib) [1]. MATE code does not need to be knowing about the different remote desktop frameworks/technologies. This will all be outsourced into librda.

These first PRs will port features provided by an X2Go Server add-on package named x2gomatebindings [2] to MATE natively. The x2gomatebindings uses some hacky-whacky approaches to implant some features / functionalities into the MATE desktop env when run on top of X2Go:

  * offer a system menu item "Suspend X2Go session..." (this PR)
  * support X2Go folder sharing in Caja natively (not yet in prep)
  * MIME type bindings for SSHFS shares mounted via X2Go (not yet in prep)
  * allow X2Go session suspension as a button in mate-screensaver-dialog (w-i-p, not in 
    x2gomatebindings)

Unfortunately, these x2gomatebindings features are--when package is installed--always present in the desktop environment, even when run locally.

NEW: The approach I'll take with my current effort (customer driven, luckily) will be generic and portable to other remote desktop frameworks than X2Go. It will also possible to mimick my patches against MATE by other desktop envs, so that also LXDE, KDE, etc. will long-term-wise benefit from this approach, I guess.

(But I'll start with X2Go for now). 

About RDA shared lib: This new shared lib will provide a technology agnostic API to remote desktop environment components like mate-panel, mate-screensaver, caja, etc. and handle the technology based stuff (how to suspend a running session in X2Go, is xRDP able to suspend sessions?, how to unmount a client-side folder in native RDP from inside the session?, etc.). All the internals of the several remote technologies will be hidden from the desktop environment itself.

E.g.: (what this PR does...)

X2Go sessions are suspendible (user can disconnect from the session and reconnect/resume later on). During suspension, an X2Go session idles on the X2Go Server in throttled mode (if not configured otherwise). When resuming, you return to the same screen arrangement (open applications, open documents, etc.) like when you left it (suspended it).

The (MATE) desktop environment programmers need to check:

```
  rda_session_can_be_suspended()
```

first and if it results as true, offer some menu item that will trigger suspension by calling

```
  rda_session_suspend()
```

For X2Go, the RDA lib will then--under the bonnet--call "x2gosuspend-session" as external command.

For a non-X2Go remote desktop environment (e.g. OgonRDP, xRDP, tigervnc, etc.) it may be that suspending of a session is not supported and surely, if supported, the internal suspension trigger will be completely different.

> Steps to test?

  * Create a test machine (one machine is enough)
  * Install X2Go Server and X2Go Client on it
  * install (build from source) librda
  * build mate-panel against it with the patch of this PR
  * launch a local MATE session
  * compare to launching an X2Go session (see below)

For launching an X2Go Session, start X2Go Client.

Configure a session profile:

```
  Host: localhost
  Port: 22 (X2Go works over SSH)
  Username: <some-user-or-same-user-on-this-test-machine>
```

Yes, you can X2Go into localhost as the same user as the one that runs the local session. You don't need two machines for testing(!!!).

> What should happen?

In the System menu of the MATE desktop, you should see no difference if the MATE session runs on local / virtual hardware directly.

If run inside an X2Go Session, there should be a "Suspend X2Go Session..." menu item under the System menu.

I will shortly add another commit to this PR, that enforces hiding the "Shutdown" menu item when run in an X2Go session (as shutting down a machine from remote via System menu makes not much sense and has sometimes caused some annoyance if it really worked -> in most cases policykit jumps in and prohibits immediate shutdown).

> What should not happen?

In X2Go Sessions, you should always have the "Suspend X2Go Session..." menu item. It should never appear, when the MATE session is running on local hardware.

> Better description please....

Yes. Thanks for demanding it. I hope things have become clearer now.

> Sorry, my crystal ball is broken :)

Yep. Sorry.

And::: if you guys have not started playing with MATE in remote desktop sessions, you should surely try it out (apt-get install x2goserver x2goclient on Debian-like systems). It is very easy and highly comfortable to have remote desktop'ing available, even on local machines.

light+love,
Mike

[1] https://github.com/ArcticaProject/librda
[2] https://code.x2go.org/gitweb?p=x2gomatebindings.git;a=summary

raveit65 commented:
@monsta 
Do you agree with adding code for a external application?

raveit65 commented:
I don't see the `suspend-x2go-session` in x2go-client. where should i see it in menus?

raveit65 commented:
Ok, i forgot to add BR librda-devel  to mate-panel.spec
But building failed against librda?
```
libtool: link: gcc -Wall -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wno-sign-compare -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wl,-z -Wl,relro -Wl,-z -Wl,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -o mate-panel-test-applets mate_panel_test_applets-panel-test-applets-resources.o mate_panel_test_applets-panel-marshal.o mate_panel_test_applets-panel-modules.o mate_panel_test_applets-panel-applet-info.o mate_panel_test_applets-panel-applets-manager.o mate_panel_test_applets-panel-test-applets.o -Wl,--export-dynamic -pthread -Wl,--export-dynamic  ../mate-panel/libmate-panel-applet-private/.libs/libmate-panel-applet-private-mini.a ../mate-panel/libpanel-util/.libs/libpanel-util.a -lgmodule-2.0 -lmate-desktop-2 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lstartup-notification-1 -lmate-menu -ldconf -lgio-2.0 -lgobject-2.0 -lglib-2.0 -pthread
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_is_ogon'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_suspend_ogon'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_terminate_ogon'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:848: mate-panel] Error 1
make[4]: *** Waiting for unfinished jobs....
Generating and caching the translation database
Merging translations into mate-panel.desktop.
make[4]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2/mate-panel'
make[3]: *** [Makefile:1716: all-recursive] Error 1
make[3]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2/mate-panel'
make[2]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2/mate-panel'
make[2]: *** [Makefile:757: all] Error 2
make[1]: *** [Makefile:536: all-recursive] Error 1
make[1]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2'
make: *** [Makefile:468: all] Error 2

```

raveit65 commented:
`--enable-introspection=yes` seems also be useless?
Because it doesn't install typical gir files.

raveit65 commented:
Same issue with a rebuilt librda.
```
%configure \
   --enable-x2go \
   --enable-introspection=no
```
```
librda 0.0.1
============

    prefix:                       /usr
    exec_prefix:                  /usr
    libdir:                       /usr/lib64
    bindir:                       /usr/bin
    sbindir:                      /usr/sbin
    sysconfdir:                   /etc
    localstatedir:                /var
    datadir:                      /usr/share
    source code location:         .
    compiler:                     gcc
    cflags:                       -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
    Use *_DISABLE_DEPRECATED:     no

    Build introspection support:  no (disabled, use --enable-introspection to enable)

    Enable X2Go awareness         yes
    Enable Ogon awareness         no
```
```
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_is_ogon'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_suspend_ogon'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_terminate_ogon'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:848: mate-panel] Error 1
```

sunweaver commented:
On  Mi 03 Okt 2018 18:41:49 CEST, raveit65 wrote:

> `--enable-introspection=yes` seems also be useless?
> Because it doesn't install typical gir files.

I will add that... Thanks for pointing it out.
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
Hi,

On  Mi 03 Okt 2018 19:16:10 CEST, raveit65 wrote:

> Same issue with a rebuilt librda.
> ```
> %configure \
>    --enable-x2go \
>    --enable-introspection=no
> ```
> ```
> librda 0.0.1
> ============
>
>     prefix:                       /usr
>     exec_prefix:                  /usr
>     libdir:                       /usr/lib64
>     bindir:                       /usr/bin
>     sbindir:                      /usr/sbin
>     sysconfdir:                   /etc
>     localstatedir:                /var
>     datadir:                      /usr/share
>     source code location:         .
>     compiler:                     gcc
>     cflags:                       -O2 -g -pipe -Wall  
> -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  
> -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong  
> -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1  
> -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic  
> -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
>     Use *_DISABLE_DEPRECATED:     no
>
>     Build introspection support:  no (disabled, use  
> --enable-introspection to enable)
>
>     Enable X2Go awareness         yes
>     Enable Ogon awareness         no
> ```
> ```
> /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so:  
> undefined reference to `rda_session_is_ogon'
> /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so:  
> undefined reference to `rda_session_suspend_ogon'
> /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so:  
> undefined reference to `rda_session_terminate_ogon'
> collect2: error: ld returned 1 exit status
> make[4]: *** [Makefile:848: mate-panel] Error 1
> ```

Does this [1] fix the above issue?

Mike

[1]  
https://github.com/ArcticaProject/librda/commit/614e86da5d2fbfdf101fb490f49465e8572c479f
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
No, same problem.
```
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_is_x2go'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_terminate_x2go'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_suspend_x2go'
collect2: error: ld returned 1 exit status
```
I think this is a linker error in make files.

sunweaver commented:
Hi Wolfgang,

On  Mi 03 Okt 2018 19:16:10 CEST, raveit65 wrote:

> Same issue with a rebuilt librda.
> ```
> %configure \
>    --enable-x2go \
>    --enable-introspection=no
> ```
> ```
> librda 0.0.1
> ============
>
>     prefix:                       /usr
>     exec_prefix:                  /usr
>     libdir:                       /usr/lib64
>     bindir:                       /usr/bin
>     sbindir:                      /usr/sbin
>     sysconfdir:                   /etc
>     localstatedir:                /var
>     datadir:                      /usr/share
>     source code location:         .
>     compiler:                     gcc
>     cflags:                       -O2 -g -pipe -Wall  
> -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2  
> -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong  
> -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1  
> -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic  
> -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
>     Use *_DISABLE_DEPRECATED:     no
>
>     Build introspection support:  no (disabled, use  
> --enable-introspection to enable)
>
>     Enable X2Go awareness         yes
>     Enable Ogon awareness         no
> ```
> ```
> /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so:  
> undefined reference to `rda_session_is_ogon'
> /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so:  
> undefined reference to `rda_session_suspend_ogon'
> /usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so:  
> undefined reference to `rda_session_terminate_ogon'
> collect2: error: ld returned 1 exit status
> make[4]: *** [Makefile:848: mate-panel] Error 1
> ```

What are doing differently from what I do???
http://paste.debian.net/1050092/

Here (on Debian stable, also tested on Debian sid) it works...

Did you try with the latest version on the master branch?

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
I am using `mock` to build rpms, which is the local client of what is used by fedora build server (koji).
Well, i can test compiling by hand (.configure && make && make install).
But this doesn't make any sence if i can't use it as official build for fedora.
And why i am the only one who tries to build this PR with librda?

sunweaver commented:
On  Fr 02 Nov 2018 13:44:22 CET, raveit65 wrote:

> I am using `mock` to build rpms, which is the local client of what  
> is used by fedora build server (koji).
> Well, i can test compiling by hand (.configure && make && make install).
> But this doesn't make any sence if i can't use it as official build  
> for fedora.
> And why i am the only one who tries to build this PR with librda?

Do you test with the 0.0.1 release? Or with a later version?

If you use an old version, please try with master/HEAD of librda and  
let me know if this works.

If so, I'll bump a new release and provide you with an up-to-date tarball.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
Hi Mike,
I created a new tarball from git, but having a new released tarball makes my life easier ;)

raveit65 commented:
I am sorry to say...
```
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_is_x2go'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_terminate_x2go'
/usr/lib/gcc/x86_64-redhat-linux/8/../../../../lib64/librda.so: undefined reference to `rda_session_suspend_x2go'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:848: mate-panel] Error 1
make[4]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2/mate-panel'
make[3]: *** [Makefile:1716: all-recursive] Error 1
make[3]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2/mate-panel'
make[2]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2/mate-panel'
make[2]: *** [Makefile:757: all] Error 2
make[1]: *** [Makefile:536: all-recursive] Error 1
make[1]: Leaving directory '/builddir/build/BUILD/mate-panel-1.21.2'
make: *** [Makefile:468: all] Error 2
Fehler: Fehler-Status beim Beenden von /var/tmp/rpm-tmp.rSc6Pu (%build)
    Fehler-Status beim Beenden von /var/tmp/rpm-tmp.rSc6Pu (%build)
```

raveit65 commented:
full mock build-log
https://www.dropbox.com/s/3swpct80ko5lpx6/mate-panel-log.tar.xz?dl=0

raveit65 commented:
Note, this is m-p from master + open a few open PRs and all is build in a chroot environment for mock.

sunweaver commented:
Hi Wolfgang,

On  Sa 03 Nov 2018 21:29:09 CET, raveit65 wrote:

> full mock build-log
> https://www.dropbox.com/s/3swpct80ko5lpx6/mate-panel-log.tar.xz?dl=0

I think I have an idea what's the problem with your build. In librda,  
I need to statically adapt the rda.h file and enable / disable the  
technologies enabled/disabled at build time of librda. Otherwise, you  
would have to set the WITH_REMOTE_TECHNOLOGY_* macros via mate-panel  
Makefile.am (and this does not make sense).

I will look into a fix for this on Monday/Tuesday.

Greets,
Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
On  Sa 03 Nov 2018 21:29:09 CET, raveit65 wrote:

> full mock build-log
> https://www.dropbox.com/s/3swpct80ko5lpx6/mate-panel-log.tar.xz?dl=0

Try latest Git master/HEAD of librda, with this [1] commit added  
(which should fix the problem)...

Mike

[1]  
https://github.com/ArcticaProject/librda/commit/d72ba23b8551ee20674c7e6e142027e77214ba1c

-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
Hi Mike, now i got this build error.
```
In file included from panel-menu-items.c:43:
/usr/include/rda/rda.h:29:10: fatal error: rda_supported_technologies.h: No such file or directory
 #include <rda_supported_technologies.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [Makefile:1323: mate_panel-panel-menu-items.o] Error 1
```
@mate-desktop/core-team 
This should really test by someone other.

raveit65 commented:
And the missing file is installed in my mock build environment.
```
[rave@mother ~]$ ls -la /home/rave/fedora-28-x86_64-mate-1.22/root/usr/include/rda
insgesamt 24
drwxr-xr-x  2 root root  4096  7. Nov 11:43 .
drwxr-xr-x 80 root root 12288  7. Nov 11:43 ..
-rw-r--r--  1 root root  1597  7. Nov 11:22 rda.h
-rw-r--r--  1 root root  1161  7. Nov 11:39 rda_supported_technologies.h
```

sunweaver commented:
Hi Wolfgang,

On  Mi 07 Nov 2018 11:55:19 CET, raveit65 wrote:

> Hi Mike, now i got this build error.
> ```
> In file included from panel-menu-items.c:43:
> /usr/include/rda/rda.h:29:10: fatal error:  
> rda_supported_technologies.h: No such file or directory
>  #include <rda_supported_technologies.h>
>           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> compilation terminated.
> make[4]: *** [Makefile:1323: mate_panel-panel-menu-items.o] Error 1
> ```
> @mate-desktop/core-team
> This should really test by someone other.

I have now tested the mate-panel build myself. With this patch for  
librda, my build of mate-panel works:
https://github.com/ArcticaProject/librda/commit/9c0f5f7570e12f0f19366a19d58ba8cd0062be59

I hope your next build attempt will work. Sorry that I have not  
re-build mate-panel here locally test-wise after the recent librda  
changes.

_o/

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
Good news, mate-panel builds without new build warnings now.
I will start to test function with related mate-screensaver PR over the weekend.

sunweaver commented:
On  Fr 09 Nov 2018 15:08:51 CET, raveit65 wrote:

> Good news, mate-panel builds without new build warnings now.
> I will start to test function with related mate-screensaver PR over  
> the weekend.

Awesome. Thanks!

Note that the mate-screensaver itself needs some more work.

When resuming from X2Go suspension, you resume to a locked screen.  
This is suboptimal. It would be cool to have some means to unlock the  
screensaver from externally (e.g. via mate-screensaver-command).  
Allowing the logged-in user a script based / cmdline based unlock  
should have no security implications IMHO (it's just a more gentle  
killall, in fact).

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
Same missing `suspend-x2go-session` entry in menus if using my notebook as x2go-server.
So for what is this PR good?

sunweaver commented:
HI Wolfgang,

On  Sa 10 Nov 2018 11:39:46 CET, raveit65 wrote:

> Same missing `suspend-x2go-session` entry in menus if using my  
> notebook as x2go-server.
> So for what is this PR good?

I released librda 0.0.3 tonight and fixed a nasty flaw in the last  
release [1]. The build omitted several of the X2Go specific functions.

No clue, why my local tests worked, probably I had a version without  
any flaw when doing my demo at the X2Go Gathering.

I did a lot of testing on Debian tonight and the new version has all  
the required symbols built-in now.

The mate-panel PR should now work.

For mate-screensaver, I did more work tonight and I will update that  
PR after the weekend.

I also added a little tool to librda called "rdacheck". It prints some  
basic info to stdout:

```
[mike@minobo ~]$ rdacheck
** Message: Currently used remote technology: X2Go
** Message: RDA supports the following remote session technologies:
** Message:     * X2Go
** Message:     * OgonRDP
```

Greets,
Mike

[1]  
https://github.com/ArcticaProject/librda/commit/3f5cd65640927c562adcb4bbe6d00b0f2a0f91b4
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

lukefromdc commented:
I do not like the idea of making remote desktop support (which many NEVER use) mandatory to build mate-panel. Please make this an optional dependency or one which autochecks for remote desktop support and simply disables it if not present

sunweaver commented:
Hi,

On  So 20 Jan 2019 01:09:00 CET, lukefromdc wrote:

> I do not like the idea of making remote desktop support (which many  
> NEVER use) mandatory to build mate-panel. Please make this an  
> optional dependency or one which autochecks for remote desktop  
> support and simply disables it if not present

RDA is indeed optional, already. If RDA is not present in the build  
environment, mate-panel will build without RDA support just fine. See  
this PR's change to configure.ac.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

sunweaver commented:
Hi Luke,

On  Di 22 Jan 2019 18:30:08 CET, lukefromdc wrote:

> I cannot easily test this as I have no remote desktop experience,  
> programs etc. Leaving this for the rest of the team

But it is really simple to test:

Install a Debian unstable virtual machine with MATE.

Then:

```
sudo apt install x2goserver
```

and on the same machine:

```
sudo apt install x2goclient
```

Then log into MATE locally. Open X2Go Client, create a new session  
profile, that connects to localhost's SSH port with session type "MATE".

Then launch the "remote" MATE/X2Go session (to localhost... not so  
remote, really).

In the local MATE session, you see:

   * shutdown menu item in System menu
   * no hint of X2Go functionalities, neither in the panel, nor in the  
screensaver

Whereas in the remote MATE session, you see:

   * no shutdown menu item
   * but a "Suspend X2Go session..." menu item
   * and a "Disconnect X2Go" button in the screensaver's log dialog

Maybe you want to give it a try (and use it elsewhere: X2Go uses SSH  
underneath, so all machines that are SSH'able can be turned into  
machines being X2Go'able).

Greets,
Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

lukefromdc commented:
I have no virtual machine experience either, plus no landline so no bandwidth for large downloads. Sorry about it, but I am not in a position to work on remote desktop and virtual machine stuff. Fortunately, we are a team and there are others here who can work on this.  I

lukefromdc commented:
Got this backtrace on the non-librda build segfault on opening the add to panel dialog. To duplicate, uninstall librda0 and librda-dev, then reconfigure, rebuild, and reinstall. Start the panel, and try to open the add to panel dialog. This is the backtrace from the resulting segfault (in "??"):
```

(gdb) thread apply all bt full

Thread 5 (Thread 0x7ffff1def700 (LWP 19260)):
#0  0x00007ffff60ba269 in syscall ()
    at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
#1  0x00007ffff6900904 in g_cond_wait_until ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff688cfd1 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff688d571 in g_async_queue_timeout_pop ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff68e2ce1 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff68e22a5 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#6  0x00007ffff761dfa3 in start_thread (arg=<optimized out>)
    at pthread_create.c:486
        ret = <optimized out>
        pd = <optimized out>
        now = <optimized out>
        unwind_buf = 
              {cancel_jmp_buf = {{jmp_buf = {140737251309312, -7886936750938171689, 140737488347374, 140737488347375, 140737251309312, 0, 7886959023971063511, 7886953090053134039}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#7  0x00007ffff60bf7ef in clone ()

    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 4 (Thread 0x7ffff2d2c700 (LWP 19239)):
#0  0x00007ffff60b4b39 in __GI___poll (fds=0x5555556a3b60, nfds=1, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
        resultvar = 18446744073709551100
        sc_cancel_oldtype = 0
#1  0x00007ffff68ba6fe in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff68ba80c in g_main_context_iteration ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff7fc6ffd in  ()
    at /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
#4  0x00007ffff68e22a5 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff761dfa3 in start_thread (arg=<optimized out>)
    at pthread_create.c:486
        ret = <optimized out>
        pd = <optimized out>
        now = <optimized out>
        unwind_buf = 
              {cancel_jmp_buf = {{jmp_buf = {140737267287808, -7886936750938171689, 140737488347342, 140737488347343, 140737267287808, 0, 7886961191318935255, 7886953090053134039}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}

        not_first_call = <optimized out>
#6  0x00007ffff60bf7ef in clone ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 3 (Thread 0x7ffff35b5700 (LWP 19238)):
#0  0x00007ffff60b4b39 in __GI___poll (fds=0x7fffe4019320, nfds=3, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
        resultvar = 18446744073709551100
        sc_cancel_oldtype = 0
#1  0x00007ffff68ba6fe in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff68baa72 in g_main_loop_run ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff6ab7186 in  () at /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00007ffff68e22a5 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff761dfa3 in start_thread (arg=<optimized out>)
    at pthread_create.c:486
        ret = <optimized out>
        pd = <optimized out>
        now = <optimized out>
        unwind_buf = 
              {cancel_jmp_buf = {{jmp_buf = {140737276237568, -7886936750938171689, 140737488348254, 140737488348255, 140737276237568, 0, 7886962353644459735, 7886953090053134039}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, d
ata = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#6  0x00007ffff60bf7ef in clone ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 2 (Thread 0x7ffff3db6700 (LWP 19237)):
#0  0x00007ffff60b4b39 in __GI___poll (fds=0x5555556047c0, nfds=2, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
        resultvar = 18446744073709551100
        sc_cancel_oldtype = 0
#1  0x00007ffff68ba6fe in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff68ba80c in g_main_context_iteration ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff68ba851 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff68e22a5 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff761dfa3 in start_thread (arg=<optimized out>)
    at pthread_create.c:486
        ret = <optimized out>
        pd = <optimized out>
        now = <optimized out>
        unwind_buf = 
              {cancel_jmp_buf = {{jmp_buf = {140737284630272, -7886936750938171689, 140737488347886, 140737488347887, 140737284630272, 0, 7886963453692958423, 7
886953090053134039}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
#6  0x00007ffff60bf7ef in clone ()
    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 1 (Thread 0x7ffff409ca80 (LWP 19233)):
#0  0x00000000000001b8 in  ()
#1  0x00005555555b3d07 in panel_addto_prepend_internal_applets
    (list=0x55555595cc50 = {...}) at panel-addto.c:234
        info = <optimized out>
        i = 8
        translated = 1
        model = <optimized out>
        l = <optimized out>
#2  0x00005555555b3d07 in panel_addto_make_applet_model (dialog=0x555555c8d000)
    at panel-addto.c:481
        model = <optimized out>
        l = <optimized out>
#3  0x00005555555b3d07 in panel_addto_present_applets
    (dialog=dialog@entry=0x555555c8d000) at panel-addto.c:861
#4  0x00005555555b4b75 in panel_addto_present
    (item=<optimized out>, panel_widget=0x5555558982d0) at panel-addto.c:1397

        dialog = 0x555555c8d000
        toplevel = <optimized out>
        pd = 0x5555556508e0
        screen = 0x555555628020
        screen_height = <optimized out>
        height = 490
#5  0x00007ffff699bbbd in g_closure_invoke ()
    at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff69aebd3 in  () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#7  0x00007ffff69b7cae in g_signal_emit_valist ()
    at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00007ffff69b8352 in g_signal_emit ()
    at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00007ffff720573a in gtk_widget_activate ()
    at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#10 0x00007ffff70e1116 in gtk_menu_shell_activate_item ()
    at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#11 0x00007ffff70e1393 in  () at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#12 0x00007ffff7253354 in  () at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#13 0x00007ffff699bdf6 in  () at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff69b7747 in g_signal_emit_valist ()
    at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff69b8352 in g_signal_emit ()

    at /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff72032b4 in  () at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#17 0x00007ffff70cc6c6 in  () at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#18 0x00007ffff70ce70b in gtk_main_do_event ()
    at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#19 0x00007ffff6ddc105 in  () at /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#20 0x00007ffff6e0bbc2 in  () at /usr/lib/x86_64-linux-gnu/libgdk-3.so.0
#21 0x00007ffff68ba51e in g_main_context_dispatch ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#22 0x00007ffff68ba788 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#23 0x00007ffff68baa72 in g_main_loop_run ()
    at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#24 0x00007ffff70cd825 in gtk_main ()
    at /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#25 0x00005555555798d9 in main (argc=<optimized out>, argv=<optimized out>)
    at main.c:199
        desktopfile = <optimized out>
        context = <optimized out>
        error = 0x0
        display = <optimized out>
        screen = 0x555555628020
        css = 0x555555863a80
        provider = 0x555555863a80

        resource = 0x5555555c55a8 "/org/mate/panel/theme/mate-panel.css"
        priority = 1

```

lukefromdc commented:
Note that the Debian builds are rda-enabled and thus will never see the segfault, they should be fine as long as the package depends on librda so the panel can be started.

lukefromdc commented:
Note that Debian already builds with this now:
https://www.phoronix.com/scan.php?page=news_item&px=MATE-Debian-Remote-Aware
so if they don't get reports of this not working we can probably get away with merging it once we fix (or drop) the non-rda build case.

sunweaver commented:
@lukefromdc: thanks for spotting this. Without much testing, I investigated the code and updated the PR branch with this change compared to the previous state:

```
diff --git a/mate-panel/panel-enums-gsettings.h b/mate-panel/panel-enums-gsettings.h
index da2bcef..7f57855 100644
--- a/mate-panel/panel-enums-gsettings.h
+++ b/mate-panel/panel-enums-gsettings.h
@@ -26,6 +26,8 @@
 #ifndef __PANEL_ENUMS_GSETTINGS_H__
 #define __PANEL_ENUMS_GSETTINGS_H__
 
+#include <config.h>
+
 #include <glib.h>
 
 G_BEGIN_DECLS
@@ -68,7 +70,9 @@ typedef enum {
        PANEL_ACTION_FORCE_QUIT,
        PANEL_ACTION_CONNECT_SERVER,
        PANEL_ACTION_SHUTDOWN,
+#ifdef HAVE_RDA
        PANEL_ACTION_SUSPEND,
+#endif
        PANEL_ACTION_LAST
 } PanelActionButtonType;
 
diff --git a/mate-panel/panel-icon-names.h b/mate-panel/panel-icon-names.h
index c887a34..30d5e14 100644
--- a/mate-panel/panel-icon-names.h
+++ b/mate-panel/panel-icon-names.h
@@ -1,3 +1,5 @@
+#include <config.h>
+
 #ifndef PANEL_ICON_NAMES_H
 #define PANEL_ICON_NAMES_H
 
@@ -33,7 +35,9 @@
 #define PANEL_ICON_SAVED_SEARCH                "folder-saved-search"
 #define PANEL_ICON_SEARCHTOOL          "system-search"
 #define PANEL_ICON_SHUTDOWN            "system-shutdown"
+#ifdef HAVE_RDA
 #define PANEL_ICON_SUSPEND             "stock_media-pause"
+#endif
 #define PANEL_ICON_THEME               "preferences-desktop-theme"
 #define PANEL_ICON_TRASH               "user-trash"
 #define PANEL_ICON_UNKNOWN             "image-missing"
```

Can you re-test the non-RDA build and see if it still segfaults? If so, I will try to re-produce your segfault here locally.

Thanks,
Mike

raveit65 commented:
No buttons with latest librda !!!
![x2go-session-nobutton](https://user-images.githubusercontent.com/961604/51759597-dde79300-20c8-11e9-926c-abdabe8e412b.jpg)

sunweaver commented:
Hi,

On Friday, 25 January 2019, raveit65 wrote:
> No buttons with latest librda !!!

In the X2Go session, open a terminal and run rdacheck (it is shipped with librda). What's its output?

Also send me a list of symbols in librda.so.0. Also, send me the /usr/include/rda/rda_supported_technologies.h

Thanks,
Mike

-- 
Sent from my Sailfish device

sunweaver commented:
Please, also provide the configure summary.

Mike

On Friday, 25 January 2019, raveit65 wrote:
> No buttons with latest librda !!!
> ![x2go-session-nobutton](https://user-images.githubusercontent.com/961604/51759597-dde79300-20c8-11e9-926c-abdabe8e412b.jpg)
> 
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/mate-desktop/mate-panel/pull/824#issuecomment-45763600

-- 
Sent from my Sailfish device

raveit65 commented:
```
             mate-panel 1.21.3
              =================

        prefix:                        /usr
        exec_prefix:                   /usr
        libdir:                        /usr/lib64
        bindir:                        /usr/bin
        sbindir:                       /usr/sbin
        sysconfdir:                    /etc
        localstatedir:                 /var
        datadir:                       /usr/share
        source code location:          .
        compiler:                      gcc
        cflags:                        -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
        Maintainer mode:               no
        Use *_DISABLE_DEPRECATED:      no
        Applets to build in-process:   (none)
        Wayland support:               yes
        X11 support:                   yes
        XRandr support:                yes
        RDA support:                   yes
        Build introspection support:   yes
        Build gtk-doc documentation:   no
```
```
[rave@f28 ~]$ rdacheck
** Message: 19:03:54.490: Currently used remote technology: 
** Message: 19:03:54.490: RDA supports the following remote technologies:
```
```
[rave@f28 ~]$ cat /usr/include/rda/rda_supported_technologies.h
/* -*- Mode: C; c-set-style: linux indent-tabs-mode: t; c-basic-offset: 8; tab-width: 8 -*- */

/* rda.h - Remote Desktop Awareness

   Copyright (C) 2018-2019 Mike Gabriel
   All rights reserved.

   The RDA Library is free software; you can redistribute it and/or
   modify it under the terms of the GNU Library General Public License as
   published by the Free Software Foundation; either version 2 of the
   License, or (at your option) any later version.

   The RDA Library is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   Library General Public License for more details.

   You should have received a copy of the GNU Library General Public
   License along with the Mate Library; see the file COPYING.LIB.  If not,
   write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
   Boston, MA  02110-1301, USA.
 */

#ifndef RDA_SUPPORTED_TECHNOLOGIES_H
#define RDA_SUPPORTED_TECHNOLOGIES_H

// @WITH_REMOTE_AWARENESS_X2GO@ not supported
// @WITH_REMOTE_AWARENESS_OGON@ not supported

#endif /* RDA_SUPPORTED_TECHNOLOGIES_H */
```
```
[rave@f28 ~]$ ldd -v /usr/lib64/librda.so.0
	linux-vdso.so.1 (0x00007ffeb0cb1000)
	libgio-2.0.so.0 => /lib64/libgio-2.0.so.0 (0x00007f6ad593d000)
	libgobject-2.0.so.0 => /lib64/libgobject-2.0.so.0 (0x00007f6ad56eb000)
	libglib-2.0.so.0 => /lib64/libglib-2.0.so.0 (0x00007f6ad53d3000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f6ad51b4000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f6ad4df5000)
	libffi.so.6 => /lib64/libffi.so.6 (0x00007f6ad4bec000)
	libgmodule-2.0.so.0 => /lib64/libgmodule-2.0.so.0 (0x00007f6ad49e8000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007f6ad47e4000)
	libpcre.so.1 => /lib64/libpcre.so.1 (0x00007f6ad4573000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f6ad435c000)
	libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f6ad4133000)
	libresolv.so.2 => /lib64/libresolv.so.2 (0x00007f6ad3f1c000)
	libmount.so.1 => /lib64/libmount.so.1 (0x00007f6ad3cc2000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f6ad3aaa000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6ad5ee4000)
	libpcre2-8.so.0 => /lib64/libpcre2-8.so.0 (0x00007f6ad3826000)
	libblkid.so.1 => /lib64/libblkid.so.1 (0x00007f6ad35d5000)
	libuuid.so.1 => /lib64/libuuid.so.1 (0x00007f6ad33ce000)
	librt.so.1 => /lib64/librt.so.1 (0x00007f6ad31c6000)

	Version information:
	/usr/lib64/librda.so.0:
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libgio-2.0.so.0:
		libresolv.so.2 (GLIBC_2.2.5) => /lib64/libresolv.so.2
		libz.so.1 (ZLIB_1.2.2) => /lib64/libz.so.1
		libpthread.so.0 (GLIBC_2.2.5) => /lib64/libpthread.so.0
		libgcc_s.so.1 (GCC_3.0) => /lib64/libgcc_s.so.1
		libgcc_s.so.1 (GCC_3.3.1) => /lib64/libgcc_s.so.1
		libmount.so.1 (MOUNT_2.19) => /lib64/libmount.so.1
		libc.so.6 (GLIBC_2.9) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.12) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libgobject-2.0.so.0:
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libglib-2.0.so.0:
		libpthread.so.0 (GLIBC_2.12) => /lib64/libpthread.so.0
		libpthread.so.0 (GLIBC_2.2.5) => /lib64/libpthread.so.0
		libc.so.6 (GLIBC_2.9) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.3) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.8) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.10) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.7) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.15) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.17) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
	/lib64/libpthread.so.0:
		ld-linux-x86-64.so.2 (GLIBC_2.2.5) => /lib64/ld-linux-x86-64.so.2
		ld-linux-x86-64.so.2 (GLIBC_PRIVATE) => /lib64/ld-linux-x86-64.so.2
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.2) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_PRIVATE) => /lib64/libc.so.6
	/lib64/libc.so.6:
		ld-linux-x86-64.so.2 (GLIBC_2.3) => /lib64/ld-linux-x86-64.so.2
		ld-linux-x86-64.so.2 (GLIBC_PRIVATE) => /lib64/ld-linux-x86-64.so.2
	/lib64/libffi.so.6:
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libgmodule-2.0.so.0:
		libdl.so.2 (GLIBC_2.2.5) => /lib64/libdl.so.2
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libpthread.so.0 (GLIBC_2.2.5) => /lib64/libpthread.so.0
	/lib64/libdl.so.2:
		ld-linux-x86-64.so.2 (GLIBC_PRIVATE) => /lib64/ld-linux-x86-64.so.2
		libc.so.6 (GLIBC_PRIVATE) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libpcre.so.1:
		libpthread.so.0 (GLIBC_2.2.5) => /lib64/libpthread.so.0
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
	/lib64/libz.so.1:
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
	/lib64/libselinux.so.1:
		libdl.so.2 (GLIBC_2.2.5) => /lib64/libdl.so.2
		ld-linux-x86-64.so.2 (GLIBC_2.3) => /lib64/ld-linux-x86-64.so.2
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.8) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.7) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
	/lib64/libresolv.so.2:
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_PRIVATE) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
	/lib64/libmount.so.1:
		librt.so.1 (GLIBC_2.2.5) => /lib64/librt.so.1
		ld-linux-x86-64.so.2 (GLIBC_2.3) => /lib64/ld-linux-x86-64.so.2
		libblkid.so.1 (BLKID_1.0) => /lib64/libblkid.so.1
		libblkid.so.1 (BLKID_2.17) => /lib64/libblkid.so.1
		libblkid.so.1 (BLKID_2.15) => /lib64/libblkid.so.1
		libc.so.6 (GLIBC_2.9) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.25) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.15) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.8) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.2) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.7) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.17) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.6) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libgcc_s.so.1:
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libpcre2-8.so.0:
		libpthread.so.0 (GLIBC_2.2.5) => /lib64/libpthread.so.0
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
	/lib64/libblkid.so.1:
		ld-linux-x86-64.so.2 (GLIBC_2.3) => /lib64/ld-linux-x86-64.so.2
		libuuid.so.1 (UUID_1.0) => /lib64/libuuid.so.1
		libc.so.6 (GLIBC_2.25) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.8) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.7) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.17) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/libuuid.so.1:
		ld-linux-x86-64.so.2 (GLIBC_2.3) => /lib64/ld-linux-x86-64.so.2
		libc.so.6 (GLIBC_2.3) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.25) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
	/lib64/librt.so.1:
		libpthread.so.0 (GLIBC_2.3.2) => /lib64/libpthread.so.0
		libpthread.so.0 (GLIBC_PRIVATE) => /lib64/libpthread.so.0
		libpthread.so.0 (GLIBC_2.2.5) => /lib64/libpthread.so.0
		libc.so.6 (GLIBC_2.14) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.3.2) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.4) => /lib64/libc.so.6
		libc.so.6 (GLIBC_2.2.5) => /lib64/libc.so.6
		libc.so.6 (GLIBC_PRIVATE) => /lib64/libc.so.6
```

sunweaver commented:
On  Fr 25 Jan 2019 19:12:28 CET, raveit65 wrote:

> [rave@f28 ~]$ cat /usr/include/rda/rda_supported_technologies.h

> [...]

> #ifndef RDA_SUPPORTED_TECHNOLOGIES_H
> #define RDA_SUPPORTED_TECHNOLOGIES_H
>
> // @WITH_REMOTE_AWARENESS_X2GO@ not supported
> // @WITH_REMOTE_AWARENESS_OGON@ not supported
>
> #endif /* RDA_SUPPORTED_TECHNOLOGIES_H */

You haven't built RDA with X2Go support...

Use --enable-x2go as configure option.

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

raveit65 commented:
I did
```
%prep
%autosetup -p1

autoreconf -fi

%build
%configure \
   --enable-x2go \
   --disable-ogon \
   --enable-introspection=yes
         
make %{?_smp_mflags} V=1

%install
%{make_install}
```

sunweaver commented:
Hi,

On Friday, 25 January 2019, raveit65 wrote:
> I did
> ```
> %prep
> %autosetup -p1
> 
> autoreconf -fi
> 
> %build
> %configure \
>    --enable-x2go \
>    --disable-ogon \
>    --enable-introspection=yes
>          
> make %{?_smp_mflags} V=1
> 
> %install
> %{make_install}
> ```

In src/Makefile.am in RDA, there is sed magic that creates src/rda_supported_technologies.h. This seems to fail on Fedora?

Or, can you send the configure summary of librda, too, please? Maybe configure.ac is failing for you.

Thanks for your patience...

Mike 

-- 
Sent from my Sailfish device

raveit65 commented:
```
librda 0.0.5
============

    prefix:                       /usr
    exec_prefix:                  /usr
    libdir:                       /usr/lib64
    bindir:                       /usr/bin
    sbindir:                      /usr/sbin
    sysconfdir:                   /etc
    localstatedir:                /var
    datadir:                      /usr/share
    source code location:         .
    compiler:                     gcc
    cflags:                       -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection
    Use *_DISABLE_DEPRECATED:     no

    Build introspection support:  yes

    Enable X2Go awareness         yes
    Enable Ogon awareness         yes
```
Weird that Ogon is enabled with ` --disable-ogon` ! This looks wrong
```
+ ./configure --build=x86_64-redhat-linux-gnu --host=x86_64-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --enable-x2go --disable-ogon --enable-introspection=yes
```
Full mock build log is here https://www.dropbox.com/s/ce9qsuv5fxjq5ei/librda-build-log?dl=0

lukefromdc commented:
Any updates on this? Note that two files conflicting w current master (cannot be auto-rebased) need to be manually fixed

lukefromdc commented:
Any updates on this? Should it be closed for lack of activity, is it still being worked on, or something else?

sunweaver commented:
@lukefromdc I have updated the update branch of this PR now. Please revisit. Thanks.

lukefromdc commented:
Thanks. Checking a build without RDA support (which I do not have installed nor know how to use). Will check wayland for any surprises there

lukefromdc reviewed (approved):
A second build with librda0/librda-dev etc installed worked fine, though I could only test it running locally. Worked fine in both x11 and in wayland however.

Note that there are remote wayland setups being worked on, though unlike x11 this is not supported by default so at some future time extending this to remote wayland sessions may come up.

I also tested running a librda build with librda removed, naturally this brought up
`error while loading shared libraries: librda.so.0: cannot open shared object file: No such file or directory`
in terminal. Distros shipping librda builds will have to include librda as a panel dependency as a result. Local builders can built to suit of course.

Approving, it's about time we get this done since there are folks who can use it and the code has been written and updated

lukefromdc commented:
I have had successful LOCAL tests of this, but we need a successful test w a remote desktop (which I am not set up for) to merge this. It's been open for years

sunweaver commented:
> I have had successful LOCAL tests of this, but we need a successful test w a remote desktop (which I am not set up for) to merge this. It's been open for years. Alternately we could close it, and reopen it if interest reappears

If you are on Debian or Ubuntu, the test is pretty easy:

```
sudo apt-get install x2goserver
sudo apt-get install pyhoca-cli
pyhoca-cli --server localhost --command MATE
```
Just tested on a Debian 12 system.

Note: You don't need a remote server to test RDA (via X2Go). You can test it all on localhost.

lukefromdc commented:
Got these results (a fail) with that test on my system, which is based on Debian Sid:

```
luke@ubuntu:~$ pyhoca-cli --server localhost --command MATE
pyhoca-cli[176252] NOTICE: preparing requested X2Go session
pyhoca-cli[176252] (PyHocaCLI) NOTICE: preparing requested X2Go session
pyhoca-cli[176252] (x2gosessregistry-pylib) NOTICE: registering X2Go session Pyhoca-Client_Session...
pyhoca-cli[176252] (x2goclient-pylib) NOTICE: initializing X2Go session...
pyhoca-cli[176252] (x2gocontrolsession-pylib) NOTICE: connecting to [localhost]:22
pyhoca-cli: error: a socket error occured while establishing the connection: [Errno None] Unable to connect to port 22 on 127.0.0.1
luke@ubuntu:~$ 
```
I will clean the packages out of my install but save them in a directory for any future tests if this is fixed

lukefromdc commented:
Any updates on this?

sunweaver commented:
> Got these results (a fail in starting an X2go session, not in mate-panel) with that test on my system, which is based on Debian Sid:
> 
> ```
> luke@ubuntu:~$ pyhoca-cli --server localhost --command MATE
> pyhoca-cli[176252] NOTICE: preparing requested X2Go session
> pyhoca-cli[176252] (PyHocaCLI) NOTICE: preparing requested X2Go session
> pyhoca-cli[176252] (x2gosessregistry-pylib) NOTICE: registering X2Go session Pyhoca-Client_Session...
> pyhoca-cli[176252] (x2goclient-pylib) NOTICE: initializing X2Go session...
> pyhoca-cli[176252] (x2gocontrolsession-pylib) NOTICE: connecting to [localhost]:22
> pyhoca-cli: error: a socket error occured while establishing the connection: [Errno None] Unable to connect to port 22 on 127.0.0.1
> luke@ubuntu:~$ 
> ```
> 
> I will clean the packages out of my install but save them in a directory for any future tests if this is fixed

Sorry for the delay in response. I did not see this message any earlier. The error above looks like the test host is not running SSH on port 22 of 127.0.0.1? X2Go communicates everything via SSH.

Mike

lukefromdc commented:
I simply don't know enough about remote work to proceed any further with testing X2go at this point, all of it is outside my experience. Someone who is more familiar with this will need to run a remote test

lukefromdc commented:
Meanwhile, we know this doesn't seem to break anything running locally

lukefromdc commented:
No response on this for almost a year. Should we close it?

lukefromdc commented:
We can reopen this if folks want to start working on it again,
